packages/cozy-app-publish/cozy-app-publish.js

```diff
@@ -107,45 +114,35 @@ function _getPublishMode() {
 
 async function publishApp(cliOptions) {
   const publishMode = _getPublishMode()
-  if (publishMode === MODES.TRAVIS) {
-    console.log()
-    console.log(`${colorize.bold('Travis')} ${colorize.blue('publish mode')}`)
-    console.log()
-    try {
-      await scripts.travis({
-        registryToken: cliOptions.token,
-        buildDir: cliOptions.buildDir,
-        buildCommit: cliOptions.buildCommit,
-        buildUrl: cliOptions.buildUrl,
-        prepublishHook: cliOptions.prepublishHook,
-        postpublishHook: cliOptions.postpublishHook,
-        registryUrl: cliOptions.registryUrl,
-        spaceName: cliOptions.space,
-        verbose: cliOptions.verbose
-      })
-    } catch (error) {
-      console.error(`↳ ❌  ${error.message}`)
-      process.exit(1)
-    }
-  } else if (publishMode === MODES.MANUAL) {
-    console.log()
-    console.log(`${colorize.bold('Manual')} ${colorize.blue('publish mode')}`)
-    console.log()
-    scripts.manual({
-      registryToken: cliOptions.token,
-      buildDir: cliOptions.buildDir,
+  const acceptedModes = Object.values(MODES).includes(publishMode)
+  if (!acceptedModes.includes(publishMode)) {
+    throw new Error(
+      `Absent or unrecognized mode, you passed ${publishMode}. Accepted modes: ${acceptedModes}.`
+    )
+  }
+
+  const publishFn = scripts[publishMode]
+
+  console.log()
+  console.log(
+    `${colorize.bold(capitalize(publishMode))} ${colorize.blue('publish mode')}`
+  )
+  console.log()
+  return publishFn(
+    pickBy({
```

Comment:
I think we don't need to use a `pickBy` here, I prefer to have the property `undefined` than missing in the options object.

Reply:
The thing is that if you pass `undefined`, default options are overriden by undefined and this is not what we want here.

Reply:
Ok I didn't go it. So could you just explain that in a comment? It would be better for code reading for later I think.

Reply:
In fact, turns out, following another refactor, that it is not needed. https://github.com/cozy/cozy-libs/pull/327/commits/91907b11fe820059b7bb9f3d9800a0402f7150c5

Thanks